arena_test.go

```diff
@@ -2,9 +2,72 @@ package capnp
 
 import (
 	"bytes"
+	"errors"
 	"testing"
+
+	"capnproto.org/go/capnp/v3/exp/bufferpool"
 )
 
+type arenaAllocTest struct {
+	name string
+
+	// Arrange
+	init func() (Arena, map[SegmentID]*Segment)
+	size Size
+
+	// Assert
+	id   SegmentID
+	data []byte
+}
+
+func (test *arenaAllocTest) run(t *testing.T, i int) {
+	arena, _ := test.init()
+	seg, addr, err := arena.Allocate(test.size, nil, nil)
+
+	if err != nil {
```

Comment:
Small thing:  can we use `github.com/stretchr/testify/require` to reduce some of this assertion boilerplate?  I know there's some code in here that doesn't use it, but we've been trying to introduce it progressively each time we touch a test file.

Reply:
Updated.